### ISSUE TITLE
The form attribute in p:menuItem is ignored

### ISSUE THREAD

**Issue by zhizhideng** [primefaces/primefaces#2358]
1) Environment
- PrimeFaces version: 6.1

2) Expected behavior
According to document, if I specify the the "form" attribute of a menuItem, it should use the specified form for submission (It works for p:ajax). 

In our case, we are using jQuery.load loading external contents for a panel, to avoid nested forms we have to remove any outer form and use the "form" attribute to specify the actual submission form.  

3) Actual behavior
If the menuItem has a parent form, it will will use that parent form for submission. If the menuItem doesn't have a parent form, a FacesException saying "MenuItem must be inside a form element" will be thrown.

4) Quick inspection

In org.primefaces.component.menu.BaseMenuRenderer#encodeMenuItem, there is null check preventing menuItem outside a form to be rendered.

```
                UIComponent form = ComponentTraversalUtils.closestForm(context, menu);
                if(form == null) {
                    throw new FacesException("MenuItem must be inside a form element");
                }
```
While in  org.primefaces.component.menu.BaseMenuRenderer#buildAjaxRequest method, the form attribute of the menuItem is not used.

```
        builder.init()
                .source(clientId)
                .process(menu, source.getProcess())
                .update(menu, source.getUpdate())
                .async(source.isAsync())
                .global(source.isGlobal())
                .delay(source.getDelay())
                .timeout(source.getTimeout())
                .partialSubmit(source.isPartialSubmit(), source.isPartialSubmitSet(), source.getPartialSubmitFilter())
                .resetValues(source.isResetValues(), source.isResetValuesSet())
                .ignoreAutoUpdate(source.isIgnoreAutoUpdate())
                .onstart(source.getOnstart())
                .onerror(source.getOnerror())
                .onsuccess(source.getOnsuccess())
                .oncomplete(source.getOncomplete())
                .params(params);
        
        if(form != null) {
            builder.form(form.getClientId(context));
        }
```

**Comment by melloware** [comment]
You are right the documentation on Page 340 of the User Guide for MenuItem says "form null String Form to serialize for an ajax request. Default is the enclosing form."

So to men your Menu always has to be wrapped in a parent form however the problem is it is not respecting the form="" on the menu item.  Let me look closer at it.

**Comment by melloware** [comment]
OK I think the documentation is wrong.  MenuItem doesn't even have a "form" attribute.

```java
public interface MenuItem extends MenuElement, Confirmable {
        
    public String getIcon();
    
    public String getIconPos();
    
    public String getTitle();
    
    public boolean shouldRenderChildren();
    
    public boolean isDisabled();
    
    public String getOnclick();
    
    public String getStyle();
    
    public String getStyleClass();
    
    public String getUrl();
    
    public String getTarget();
    
    public String getOutcome();
    
    public String getFragment();

    public boolean isIncludeViewParams();
    
    public boolean isAjax();
    
    public Object getValue();
    
    public void setStyleClass(String styleClass);
    
    public Map<String, List<String>> getParams();
    
    public void setParam(String key, Object value);
    
    public boolean isDynamic();
    
    public String getCommand();
    
    public boolean isImmediate();
    
    public String getClientId();
    
    public String getContainerStyle();
    
    public String getContainerStyleClass();
    
    public boolean isEscape();
    
    public String getRel();
    
}
```

**Comment by zhizhideng** [comment]
@melloware Thanks to reply.

As I mentioned we are using jQuery.load to insert panel content from another primefaces project. The injected content has to have its own form so that the request goes to the right server. It means that the outer panel itself cannot be enclosed in a form (Some browsers do not support nested forms at all). 

The menu is in the "option" facet of the panel. I tried to wrap the menu with a form inside the option facet, although it did not crash, there was no "form" element in the final response output.

As the p:ajax has already support the "form" attribute without enclosing form, I think it would make sense to let the menuItem to support the same.

I also tried putting a p:ajax (onclick) inside the menuItem, it simply got ignored.
